Add GloParT inference facility

colizz opened the pull request:
#### PR description:

This PR adds the GloParT 3 model's inference facility into CMSSW, prepared for NanoAODv15, replacing #45830. 
Model performance details are provided in these [[slides]](https://cernbox.cern.ch/files/spaces/eos/user/c/coli/slides/24.10.18_GloParT%203%20review.pdf) (accessible via CMS).

Global Particle Transformer (GloParT) 3 is an inclusive tagging model for AK8 jets that covers the entire phase space and enables resonance mass regression for each jet class. It functions as both a global tagger and a mass regression model for AK8 jets and can also be utilized as a pre-trained model. The hidden layer neurons (with dimension 256) are stored in MiniAOD, providing the capability to resume all output scores. Further details can be found in the slides.

In summary, the integration overview (the new AK8 jet features added to MiniAOD and NanoAOD) is illustrated below.

<img width="600" alt="glopart-integration" src="https://github.com/user-attachments/assets/b33106e5-eafd-41b7-b510-1ce9830c212c">

Please test this PR with https://github.com/cms-data/RecoBTag-Combined/pull/62.


#### PR validation:

 - The PR passes the unit test `test_globalpart_cfg.py` and is also validated under the AODSIM->MINIAOD->NANOAOD workflow.
 - Profiling performance in cmssw is delivered and shown in the slides above (see pp.89-93).
 - The output scores produced via ONNX in the cmssw workflow are validated to match those in the PyTorch workflow.

colizz committed:
Add GloParT inference facility

colizz committed:
Adapt to nano workflow

cmsbuild commented:
cms-bot internal usage<!-- bot cache: {"commits":{"5f5b5c29c1b3a9d38e5e4f9a65a8df21dd14e0ec":{"files":["PhysicsTools/NanoAOD/python/jetsAK8_cff.py","PhysicsTools/PatAlgos/python/recoLayer0/bTagging_cff.py","PhysicsTools/PatAlgos/python/slimming/applyDeepBtagging_cff.py","PhysicsTools/PatAlgos/python/tools/jetTools.py","RecoBTag/Configuration/python/RecoBTag_cff.py","RecoBTag/FeatureTools/plugins/GlobalParticleTransformerAK8TagInfoProducer.cc","RecoBTag/ONNXRuntime/python/pfGlobalParticleTransformerAK8_cff.py","RecoBTag/ONNXRuntime/test/test_globalpart_cfg.py"],"squashed":false,"time":1730063100},"97b2f61cbd86b30d59502479e88a767d12931921":{"files":["PhysicsTools/NanoAOD/python/custom_jme_cff.py","PhysicsTools/NanoAOD/python/jetsAK8_cff.py","PhysicsTools/NanoAOD/python/nano_cff.py","PhysicsTools/PatAlgos/python/tools/puppiJetMETReclusteringFromMiniAOD_cff.py"],"squashed":false,"time":1730099950}},"emoji":{},"last_seen_sha":"97b2f61cbd86b30d59502479e88a767d12931921","signatures":{}} -->

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-46523/42384

cmsbuild commented:
A new Pull Request was created by  @colizz for master.

It involves the following packages:

- PhysicsTools/NanoAOD (**xpog**)
- PhysicsTools/PatAlgos (**reconstruction, xpog**)
- RecoBTag/Configuration (**reconstruction**)
- RecoBTag/FeatureTools (**reconstruction**)
- RecoBTag/ONNXRuntime (**reconstruction**)


@cmsbuild, @ftorrresd, @hqucms, @jfernan2, @mandrenguyen can you please review it and eventually sign? Thanks.
@AlexDeMoor, @AnnikaStein, @Ming-Yan, @Senphy, @ahinzmann, @andrzejnovak, @azotz, @castaned, @gkasieczka, @gouskos, @gpetruc, @hatakeyamak, @hqucms, @jdamgov, @jdolen, @mariadalfonso, @mbluj, @missirol, @mmarionncern, @nhanvtran, @rappoccio, @schoef, @seemasharmafnal this is something you requested to watch as well.
@antoniovilela, @mandrenguyen, @rappoccio, @sextonkennedy you are the release manager for this.

cms-bot commands are listed <a href="http://cms-sw.github.io/cms-bot-cmssw-cmds.html">here</a>

jfernan2 commented:
enable nano

cmsbuild commented:
-1

**Failed Tests**: UnitTests RelVals RelVals-INPUT RelVals-NANO 
**Size**: This PR adds an extra 48KB to repository
**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-d54517/42415/summary.html
**COMMIT**: 97b2f61cbd86b30d59502479e88a767d12931921
**CMSSW**: CMSSW_14_2_X_2024-10-27-2300/el8_amd64_gcc12
**Additional Tests**: NANO
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week1/cms-sw/cmssw/46523/42415/install.sh` to create a dev area with all the needed externals and cmssw changes.


## Unit Tests

I found 4 errors in the following unit tests:

<pre>---> test test_MC_22_crosscheck had ERRORS
---> test test_MC_23_crosscheck had ERRORS
---> test test_MC_22_setup had ERRORS
and more ...
</pre>

## RelVals

- [4.53](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-d54517/42415/runTheMatrix-results/4.53_RunPhoton2012B)
```
----- Begin Fatal Exception 28-Oct-2024 11:54:34 CET-----------------------
An exception of category 'ConfigFileReadError' occurred while
   [0] Processing the python configuration file named step3_RAW2DIGI_L1Reco_RECO_PAT_ALCA_DQM.py
Exception Message:
 unknown python problem occurred.
RuntimeError: An exception of category 'FileInPathError' occurred.
Exception Message:
edm::FileInPath unable to find file RecoBTag/Combined/data/GlobalParticleTransformerAK8/PUPPI/V03/model.onnx anywhere in the search path.
The search path is defined by: CMSSW_SEARCH_PATH
${CMSSW_SEARCH_PATH} is: /cvmfs/cms-ci.cern.ch/week1/cms-sw/cmssw/46523/42415/CMSSW_14_2_X_2024-10-27-2300/poison:/cvmfs/cms-ci.cern.ch/week1/cms-sw/cmssw/46523/42415/CMSSW_14_2_X_2024-10-27-2300/src:/cvmfs/cms-ci.cern.ch/week1/cms-sw/cmssw/46523/42415/CMSSW_14_2_X_2024-10-27-2300/external/el8_amd64_gcc12/data:/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02861/el8_amd64_gcc12/cms/cmssw-patch/CMSSW_14_2_X_2024-10-27-2300/poison:/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02861/el8_amd64_gcc12/cms/cmssw-patch/CMSSW_14_2_X_2024-10-27-2300/src:/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02861/el8_amd64_gcc12/cms/cmssw-patch/CMSSW_14_2_X_2024-10-27-2300/external/el8_amd64_gcc12/data
Current directory is: /data/cmsbld/jenkins/workspace/ib-run-pr-relvals/matrix-results/4.53_RunPhoton2012B


At:
  /cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02861/el8_amd64_gcc12/cms/cmssw-patch/CMSSW_14_2_X_2024-10-27-2300/src/FWCore/ParameterSet/python/Types.py(1001): insertInto
  /cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02861/el8_amd64_gcc12/cms/cmssw-patch/CMSSW_14_2_X_2024-10-27-2300/src/FWCore/ParameterSet/python/Mixins.py(397): insertContentsInto
  /cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02861/el8_amd64_gcc12/cms/cmssw-patch/CMSSW_14_2_X_2024-10-27-2300/src/FWCore/ParameterSet/python/Mixins.py(532): insertInto
  /cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02861/el8_amd64_gcc12/cms/cmssw-patch/CMSSW_14_2_X_2024-10-27-2300/src/FWCore/ParameterSet/python/Modules.py(161): insertInto
  /cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02861/el8_amd64_gcc12/cms/cmssw-patch/CMSSW_14_2_X_2024-10-27-2300/src/FWCore/ParameterSet/python/Config.py(1217): _insertManyInto
  /cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02861/el8_amd64_gcc12/cms/cmssw-patch/CMSSW_14_2_X_2024-10-27-2300/src/FWCore/ParameterSet/python/Config.py(1490): fillProcessDesc
  <string>(2): <module>

----- End Fatal Exception -------------------------------------------------
```
- [135.4](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-d54517/42415/runTheMatrix-results/135.4_ZEEFS_13)
```
----- Begin Fatal Exception 28-Oct-2024 11:54:47 CET-----------------------
An exception of category 'ConfigFileReadError' occurred while
   [0] Processing the python configuration file named step3_PAT.py
Exception Message:
 unknown python problem occurred.
RuntimeError: An exception of category 'FileInPathError' occurred.
Exception Message:
edm::FileInPath unable to find file RecoBTag/Combined/data/GlobalParticleTransformerAK8/PUPPI/V03/model.onnx anywhere in the search path.
The search path is defined by: CMSSW_SEARCH_PATH
${CMSSW_SEARCH_PATH} is: /cvmfs/cms-ci.cern.ch/week1/cms-sw/cmssw/46523/42415/CMSSW_14_2_X_2024-10-27-2300/poison:/cvmfs/cms-ci.cern.ch/week1/cms-sw/cmssw/46523/42415/CMSSW_14_2_X_2024-10-27-2300/src:/cvmfs/cms-ci.cern.ch/week1/cms-sw/cmssw/46523/42415/CMSSW_14_2_X_2024-10-27-2300/external/el8_amd64_gcc12/data:/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02861/el8_amd64_gcc12/cms/cmssw-patch/CMSSW_14_2_X_2024-10-27-2300/poison:/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02861/el8_amd64_gcc12/cms/cmssw-patch/CMSSW_14_2_X_2024-10-27-2300/src:/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02861/el8_amd64_gcc12/cms/cmssw-patch/CMSSW_14_2_X_2024-10-27-2300/external/el8_amd64_gcc12/data
Current directory is: /data/cmsbld/jenkins/workspace/ib-run-pr-relvals/matrix-results/135.4_ZEEFS_13


At:
  /cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02861/el8_amd64_gcc12/cms/cmssw-patch/CMSSW_14_2_X_2024-10-27-2300/src/FWCore/ParameterSet/python/Types.py(1001): insertInto
  /cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02861/el8_amd64_gcc12/cms/cmssw-patch/CMSSW_14_2_X_2024-10-27-2300/src/FWCore/ParameterSet/python/Mixins.py(397): insertContentsInto
  /cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02861/el8_amd64_gcc12/cms/cmssw-patch/CMSSW_14_2_X_2024-10-27-2300/src/FWCore/ParameterSet/python/Mixins.py(532): insertInto
  /cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02861/el8_amd64_gcc12/cms/cmssw-patch/CMSSW_14_2_X_2024-10-27-2300/src/FWCore/ParameterSet/python/Modules.py(161): insertInto
  /cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02861/el8_amd64_gcc12/cms/cmssw-patch/CMSSW_14_2_X_2024-10-27-2300/src/FWCore/ParameterSet/python/Config.py(1217): _insertManyInto
  /cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02861/el8_amd64_gcc12/cms/cmssw-patch/CMSSW_14_2_X_2024-10-27-2300/src/FWCore/ParameterSet/python/Config.py(1490): fillProcessDesc
  <string>(2): <module>

----- End Fatal Exception -------------------------------------------------
```
- [140.023](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-d54517/42415/runTheMatrix-results/140.023_RunZeroBias2022B)
```
----- Begin Fatal Exception 28-Oct-2024 11:55:30 CET-----------------------
An exception of category 'ConfigFileReadError' occurred while
   [0] Processing the python configuration file named step3_RAW2DIGI_L1Reco_RECO_PAT_NANO_DQM.py
Exception Message:
 unknown python problem occurred.
RuntimeError: An exception of category 'FileInPathError' occurred.
Exception Message:
edm::FileInPath unable to find file RecoBTag/Combined/data/GlobalParticleTransformerAK8/PUPPI/V03/model.onnx anywhere in the search path.
The search path is defined by: CMSSW_SEARCH_PATH
${CMSSW_SEARCH_PATH} is: /cvmfs/cms-ci.cern.ch/week1/cms-sw/cmssw/46523/42415/CMSSW_14_2_X_2024-10-27-2300/poison:/cvmfs/cms-ci.cern.ch/week1/cms-sw/cmssw/46523/42415/CMSSW_14_2_X_2024-10-27-2300/src:/cvmfs/cms-ci.cern.ch/week1/cms-sw/cmssw/46523/42415/CMSSW_14_2_X_2024-10-27-2300/external/el8_amd64_gcc12/data:/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02861/el8_amd64_gcc12/cms/cmssw-patch/CMSSW_14_2_X_2024-10-27-2300/poison:/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02861/el8_amd64_gcc12/cms/cmssw-patch/CMSSW_14_2_X_2024-10-27-2300/src:/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02861/el8_amd64_gcc12/cms/cmssw-patch/CMSSW_14_2_X_2024-10-27-2300/external/el8_amd64_gcc12/data
Current directory is: /data/cmsbld/jenkins/workspace/ib-run-pr-relvals/matrix-results/140.023_RunZeroBias2022B


At:
  /cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02861/el8_amd64_gcc12/cms/cmssw-patch/CMSSW_14_2_X_2024-10-27-2300/src/FWCore/ParameterSet/python/Types.py(1001): insertInto
  /cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02861/el8_amd64_gcc12/cms/cmssw-patch/CMSSW_14_2_X_2024-10-27-2300/src/FWCore/ParameterSet/python/Mixins.py(397): insertContentsInto
  /cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02861/el8_amd64_gcc12/cms/cmssw-patch/CMSSW_14_2_X_2024-10-27-2300/src/FWCore/ParameterSet/python/Mixins.py(532): insertInto
  /cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02861/el8_amd64_gcc12/cms/cmssw-patch/CMSSW_14_2_X_2024-10-27-2300/src/FWCore/ParameterSet/python/Modules.py(161): insertInto
  /cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02861/el8_amd64_gcc12/cms/cmssw-patch/CMSSW_14_2_X_2024-10-27-2300/src/FWCore/ParameterSet/python/Config.py(1217): _insertManyInto
  /cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02861/el8_amd64_gcc12/cms/cmssw-patch/CMSSW_14_2_X_2024-10-27-2300/src/FWCore/ParameterSet/python/Config.py(1490): fillProcessDesc
  <string>(2): <module>

----- End Fatal Exception -------------------------------------------------
```
<details>
<summary>Expand to see more relval errors ...</summary>

- [139.001](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-d54517/42415/runTheMatrix-results/139.001_RunMinimumBias2021)
- [140.063](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-d54517/42415/runTheMatrix-results/140.063_RunZeroBias2022D)
- [140.043](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-d54517/42415/runTheMatrix-results/140.043_RunZeroBias2022C)
- [136.731](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-d54517/42415/runTheMatrix-results/136.731_RunSinglePh2016B)
- [136.874](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-d54517/42415/runTheMatrix-results/136.874_RunEGamma2018C)
- [136.793](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-d54517/42415/runTheMatrix-results/136.793_RunDoubleEG2017C)
- [9.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-d54517/42415/runTheMatrix-results/9.0_Higgs200ChargedTaus)
- [25.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-d54517/42415/runTheMatrix-results/25.0_TTbar)
- [1306.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-d54517/42415/runTheMatrix-results/1306.0_SingleMuPt1_UP15)
- [1330.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-d54517/42415/runTheMatrix-results/1330.0_ZMM_13)
- [141.042](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-d54517/42415/runTheMatrix-results/141.042_RunZeroBias2023D)
- [141.044](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-d54517/42415/runTheMatrix-results/141.044_RunJetMET2023D)
- [141.046](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-d54517/42415/runTheMatrix-results/141.046_RunEGamma2023D)
- [13234.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-d54517/42415/runTheMatrix-results/13234.0_TTbar_14TeV+2021FS)
- [14034.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-d54517/42415/runTheMatrix-results/14034.0_TTbar_14TeV+2023FS)
- [25202.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-d54517/42415/runTheMatrix-results/25202.0_TTbar_13)
- [11634.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-d54517/42415/runTheMatrix-results/11634.0_TTbar_14TeV+2021)
- [14234.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-d54517/42415/runTheMatrix-results/14234.0_TTbar_14TeV+2023FSPU)
- [12434.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-d54517/42415/runTheMatrix-results/12434.0_TTbar_14TeV+2023)
- [10224.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-d54517/42415/runTheMatrix-results/10224.0_TTbar_13+2017PU)
- [12834.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-d54517/42415/runTheMatrix-results/12834.0_TTbar_14TeV+2024)
- [12846.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-d54517/42415/runTheMatrix-results/12846.0_ZEE_14+2024)
- [12834.7](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-d54517/42415/runTheMatrix-results/12834.7_TTbar_14TeV+2024_trackingMkFit)
- [29700.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-d54517/42415/runTheMatrix-results/29700.0_CloseByPGun_CE_H_Coarse_Scint+2026D110)
- [29696.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-d54517/42415/runTheMatrix-results/29696.0_CloseByPGun_CE_E_Front_120um+2026D110)
- [29634.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-d54517/42415/runTheMatrix-results/29634.0_TTbar_14TeV+2026D110)
- [13034.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-d54517/42415/runTheMatrix-results/13034.0_TTbar_14TeV+2024PU)
- [250202.181](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-d54517/42415/runTheMatrix-results/250202.181_TTbar13TeVPUppmx2018)
- [29834.999](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-d54517/42415/runTheMatrix-results/29834.999_TTbar_14TeV+2026D110PU_PMXS1S2PR)
- [24834.911](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-d54517/42415/runTheMatrix-results/24834.911_TTbar_14TeV+2026D98_DD4hep)
- [29634.911](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-d54517/42415/runTheMatrix-results/29634.911_TTbar_14TeV+2026D110_DD4hep)
</details>



## RelVals-INPUT

- [4.6](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-d54517/42415/runTheMatrixINPUT-results/4.6_MinimumBias2010A)```4.6_MinimumBias2010A/step2_MinimumBias2010A.log```
- [136.901](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-d54517/42415/runTheMatrixINPUT-results/136.901_RunDoubleMuon2016C)```136.901_RunDoubleMuon2016C/step2_RunDoubleMuon2016C.log```
- [136.902](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-d54517/42415/runTheMatrixINPUT-results/136.902_RunDoubleMuon2016H)```136.902_RunDoubleMuon2016H/step2_RunDoubleMuon2016H.log```
<details>
<summary>Expand to see more relval errors ...</summary>

- [136.903](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-d54517/42415/runTheMatrixINPUT-results/136.903_RunDoubleMuon2017B)
- [136.904](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-d54517/42415/runTheMatrixINPUT-results/136.904_RunDoubleMuon2018C)
- [2022.101001](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-d54517/42415/runTheMatrixINPUT-results/2022.101001_RunEGamma2022C_10k)
- [134.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-d54517/42415/runTheMatrixINPUT-results/134.0_MinimumBiasBS)
- [2500.001](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-d54517/42415/runTheMatrixINPUT-results/2500.001_NANOmc106Xul16v2)
- [2500.002](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-d54517/42415/runTheMatrixINPUT-results/2500.002_NANOmc106Xul17v2)
- [2500.003](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-d54517/42415/runTheMatrixINPUT-results/2500.003_NANOmc106Xul18v2)
- [2500.011](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-d54517/42415/runTheMatrixINPUT-results/2500.011_NANOdata106Xul16v2)
- [2500.012](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-d54517/42415/runTheMatrixINPUT-results/2500.012_NANOdata106Xul17v2)
- [2500.013](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-d54517/42415/runTheMatrixINPUT-results/2500.013_NANOdata106Xul18v2)
- [144.6](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-d54517/42415/runTheMatrixINPUT-results/144.6_MinimumBias2010A)
- [1002.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-d54517/42415/runTheMatrixINPUT-results/1002.0_RRD)
- [2500.021](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-d54517/42415/runTheMatrixINPUT-results/2500.021_NANOmcUL16APVreMINI)
- [2500.022](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-d54517/42415/runTheMatrixINPUT-results/2500.022_NANOmcUL16reMINI)
- [2500.023](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-d54517/42415/runTheMatrixINPUT-results/2500.023_NANOmcUL17reMINI)
- [2500.225](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-d54517/42415/runTheMatrixINPUT-results/2500.225_jmeNANOrePuppimc140X)
- [2500.031](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-d54517/42415/runTheMatrixINPUT-results/2500.031_NANOdataUL16APVreMINI)
- [2500.024](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-d54517/42415/runTheMatrixINPUT-results/2500.024_NANOmcUL18reMINI)
- [2500.032](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-d54517/42415/runTheMatrixINPUT-results/2500.032_NANOdataUL16reMINI)
- [2500.033](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-d54517/42415/runTheMatrixINPUT-results/2500.033_NANOdataUL17reMINI)
- [2500.235](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-d54517/42415/runTheMatrixINPUT-results/2500.235_jmeNANOrePuppidata140Xrun3)
- [2500.034](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-d54517/42415/runTheMatrixINPUT-results/2500.034_NANOdataUL18reMINI)
- 2022.101001
</details>



## RelVals-NANO

- [2500.003](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-d54517/42415/runTheMatrixNANO-results/2500.003_NANOmc106Xul18v2)```2500.003_NANOmc106Xul18v2/step2_NANOmc106Xul18v2.log```
- [2500.001](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-d54517/42415/runTheMatrixNANO-results/2500.001_NANOmc106Xul16v2)```2500.001_NANOmc106Xul16v2/step2_NANOmc106Xul16v2.log```
- [2500.002](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-d54517/42415/runTheMatrixNANO-results/2500.002_NANOmc106Xul17v2)```2500.002_NANOmc106Xul17v2/step2_NANOmc106Xul17v2.log```
<details>
<summary>Expand to see more relval errors ...</summary>

- [2500.012](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-d54517/42415/runTheMatrixNANO-results/2500.012_NANOdata106Xul17v2)
- [2500.011](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-d54517/42415/runTheMatrixNANO-results/2500.011_NANOdata106Xul16v2)
- [2500.013](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-d54517/42415/runTheMatrixNANO-results/2500.013_NANOdata106Xul18v2)
- [2500.031](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-d54517/42415/runTheMatrixNANO-results/2500.031_NANOdataUL16APVreMINI)
- [2500.032](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-d54517/42415/runTheMatrixNANO-results/2500.032_NANOdataUL16reMINI)
- [2500.033](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-d54517/42415/runTheMatrixNANO-results/2500.033_NANOdataUL17reMINI)
- [2500.034](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-d54517/42415/runTheMatrixNANO-results/2500.034_NANOdataUL18reMINI)
- [2500.022](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-d54517/42415/runTheMatrixNANO-results/2500.022_NANOmcUL16reMINI)
- [2500.021](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-d54517/42415/runTheMatrixNANO-results/2500.021_NANOmcUL16APVreMINI)
- [2500.023](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-d54517/42415/runTheMatrixNANO-results/2500.023_NANOmcUL17reMINI)
- [2500.024](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-d54517/42415/runTheMatrixNANO-results/2500.024_NANOmcUL18reMINI)
- [2500.225](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-d54517/42415/runTheMatrixNANO-results/2500.225_jmeNANOrePuppimc140X)
- [2500.235](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-d54517/42415/runTheMatrixNANO-results/2500.235_jmeNANOrePuppidata140Xrun3)
</details>

colizz commented:
I think it's caused by not testing with https://github.com/cms-data/RecoBTag-Combined/pull/62
Could you help test again including this test parameter? Thanks

hqucms commented:
test parameters:
 - pull_request = https://github.com/cms-data/RecoBTag-Combined/pull/62
 - enable = nano,profiling

hqucms commented:
please test

hqucms commented on RecoBTag/FeatureTools/plugins/GlobalParticleTransformerAK8TagInfoProducer.cc:
```diff
@@ -0,0 +1,742 @@
+    auto cand = lts_->ptrAt(i);
+    if (reco::deltaR(*cand, jet) < jet_radius_) {
+      cpfPtrs.push_back(cand);
+      isLostTrackMap[cand.key()] = true;
```
@colizz Is `cand.key()` unique for lost tracks and packed candidates? Or the `key()` is just the index in each collection (then different types of objects can have the same keys)? Could you please check?

colizz replied:
hi @hqucms , they are unique keys for `reco::CandidatePtr` hence should be different for each lost track and packed candidates. 
<details>
  <summary>Here is my check:</summary>
```
cpf key: 37
cpf key: 39
cpf key: 42
cpf key: 46
cpf key: 249
cpf key: 1004
cpf key: 1012
cpf key: 1013
cpf key: 1014
cpf key: 1016
cpf key: 1017
cpf key: 1018
cpf key: 1019
cpf key: 1020
cpf key: 1024
cpf key: 1025
cpf key: 1028
cpf key: 1029
cpf key: 1030
cpf key: 1032
cpf key: 1036
cpf key: 1039
cpf key: 1040
cpf key: 1041
cpf key: 1042
cpf key: 1044
cpf key: 1051
cpf key: 1052
cpf key: 1304
cpf key: 44
cpf key: 50
cpf key: 51
cpf key: 1022
cpf key: 1023
cpf key: 1026
cpf key: 1027
cpf key: 1031
cpf key: 1033
cpf key: 1034
cpf key: 1035
cpf key: 1037
cpf key: 1038
cpf key: 1043
cpf key: 1045
cpf key: 1046
cpf key: 1049
cpf key: 1056
cpf key: 1069
cpf key: 1219
lts key: 16
lts key: 17
lts key: 47
=== end of jets ===
cpf key: 47
cpf key: 56
cpf key: 978
cpf key: 1001
cpf key: 1003
cpf key: 1005
cpf key: 1053
cpf key: 1055
cpf key: 1057
cpf key: 1058
cpf key: 1059
cpf key: 1060
cpf key: 1061
cpf key: 1062
cpf key: 1063
cpf key: 1064
cpf key: 1065
cpf key: 1066
cpf key: 1068
cpf key: 1134
cpf key: 57
cpf key: 998
cpf key: 1067
cpf key: 1071
cpf key: 1072
cpf key: 1073
cpf key: 1074
cpf key: 1075
cpf key: 1076
cpf key: 1048
cpf key: 1054
cpf key: 1070
cpf key: 387
lts key: 7
lts key: 10
lts key: 12
lts key: 13
lts key: 14
lts key: 19
lts key: 49
=== end of jets ===
```
</details>

colizz replied:
thanks a lot @hqucms , this is very nice catch. There is an overlap between "keys" in PF candidates and lost tracks, with a prob of ~0.5% (though small, it indicates a logical error in the code).
We should point out that in training GloParT, we used the same code to generate training samples, which means that a small portion of charged PF candidates were mistakenly labelled as lost tracks. For now, it appears that this error is consistent across both, allowing our trained GloParT model to perform the same in CMSSW inference.

It seems our best option is to retain this code to preserve GloParT's performance. Perhaps we could add a comment here for clarity. What do you think?

hqucms replied:
@colizz Given the small possibility of the overlap, I would expect fixing this should have a rather negligible effect on the performance? If that is the indeed case I think it's better to fix it now already.

colizz replied:
ok, we will check whether the effect is negligible if we change the code here.

colizz replied:
hi @hqucms , the code has been fixed to use `CandidatePtr` itself as the map key: https://github.com/cms-sw/cmssw/pull/46523/commits/b813fb6bcb21869258145de96d49fa20136bc613
Additionally, `puppi_wgt_cache` had a similar issue; however, testing revealed that it does not produce the same errors as `isLostTrackMap`. In this updated code, we validated the scores and found that in <1% of cases, some charged PF cands are marked as `isLostTrack=True`, but this has only a minimal impact on the GloParT output scores. We also examined the ROC curve on a large sample set, and found no change to the ROC performance. Thus we conclude that this modification has a negligible impact on GloParT performance

hqucms replied:
Thanks a lot @colizz for the fast turnaround! The fix looks good to me.

nurfikri89 commented:
@colizz Thanks for making the PR. I think you need to add the GloParT3 branches in `nanoDQM_cfi.py` also.

hqucms commented:
please abort

hqucms commented:
please test

hqucms commented:
please abort

hqucms commented:
please test

colizz committed:
Fix tag info producer

 - avoid duplication between PF cands and lost tracks when using CandidatePtr key as map key

colizz committed:
Add GloParT 3 outputs to nanoDQM

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-46523/42434

- There are other open Pull requests which might conflict with changes you have proposed:
  - File  PhysicsTools/NanoAOD/python/nanoDQM_cfi.py  modified in PR(s): #46273
  - File  PhysicsTools/NanoAOD/python/nano_cff.py  modified in PR(s): #45207

cmsbuild commented:
Pull request #46523 was updated. @cmsbuild, @ftorrresd, @hqucms, @jfernan2, @mandrenguyen can you please check and sign again.

colizz commented:
> @colizz Thanks for making the PR. I think you need to add the GloParT3 branches in `nanoDQM_cfi.py` also.

@nurfikri89 thanks! I updated `nanoDQM_cfi.py` to include the new branches: https://github.com/cms-sw/cmssw/pull/46523/commits/243edc69092488ae4ed876479364b2cefa979658

hqucms commented:
please abort

hqucms commented:
please test

nurfikri89 commented:
> > @colizz Thanks for making the PR. I think you need to add the GloParT3 branches in `nanoDQM_cfi.py` also.
> 
> @nurfikri89 thanks! I updated `nanoDQM_cfi.py` to include the new branches: https://github.com/cms-sw/cmssw/pull/46523/commits/243edc69092488ae4ed876479364b2cefa979658

Brilliant. Looks good. Thanks!

hqucms commented:
@smuzaffar There seems to be some trouble in getting the tests to run through for this PR. Could you please have a look?

smuzaffar commented:
@hqucms , the profiling jobs for this PR are still in [queue](https://cmssdt.cern.ch/jenkins/job/ib-run-pr-profiling/) ( waiting for resources to be available)

hqucms commented:
> @hqucms , the profiling jobs for this PR are still in [queue](https://cmssdt.cern.ch/jenkins/job/ib-run-pr-profiling/) ( waiting for resources to be available)

I see. Thanks a lot!

cmsbuild commented:
-1

**Failed Tests**: UnitTests RelVals-INPUT 
**Size**: This PR adds an extra 36KB to repository
**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-d54517/42478/summary.html
**COMMIT**: 243edc69092488ae4ed876479364b2cefa979658
**CMSSW**: CMSSW_14_2_X_2024-10-30-1100/el8_amd64_gcc12
**Additional Tests**: NANO,PROFILING
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week1/cms-sw/cmssw/46523/42478/install.sh` to create a dev area with all the needed externals and cmssw changes.



The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
 - @JHiltbrand cms-sw/cmssw#46538
 - @smuzaffar cms-sw/cmssw#46442
 - @DebabrataBhowmik cms-sw/cmssw#46540
 - @mmusich cms-sw/cmssw#46530
 - @24LopezR cms-sw/cmssw#46529

You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-d54517/42478/git-recent-commits.json
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-d54517/42478/git-merge-result


## Unit Tests

I found 27 errors in the following unit tests:

<pre>---> test runtestPhysicsToolsPatAlgos had ERRORS
---> test test-runTheMatrix-interactive had ERRORS
---> test test_MC_22_crosscheck had ERRORS
and more ...
</pre>

## RelVals-INPUT

- [2024.000001](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-d54517/42478/runTheMatrixINPUT-results/2024.000001_RunJetMET02024D_10k)```2024.000001_RunJetMET02024D_10k/step1_dasquery.log```
- [2024.001001](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-d54517/42478/runTheMatrixINPUT-results/2024.001001_RunZeroBias2024D_10k)```2024.001001_RunZeroBias2024D_10k/step1_dasquery.log```
- [2024.100001](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-d54517/42478/runTheMatrixINPUT-results/2024.100001_RunJetMET02024C_10k)```2024.100001_RunJetMET02024C_10k/step1_dasquery.log```
<details>
<summary>Expand to see more relval errors ...</summary>

- [2024.101001](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-d54517/42478/runTheMatrixINPUT-results/2024.101001_RunZeroBias2024C_10k)
- 2024.000001
- 2024.001001
- 2024.100001
- 2024.101001
</details>



## Comparison Summary



**Summary**:
 - You potentially [added 17 lines](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_14_2_X_2024-10-30-1100+d54517/65378/validateJR/logRootQA-events.log) to the logs
 - ROOTFileChecks: Some differences in event products or their sizes found
 - [Reco comparison results: 4613 differences found in the comparisons](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_14_2_X_2024-10-30-1100+d54517/65378/validateJR.html)
 - DQMHistoTests: Total files compared: 46
 - DQMHistoTests: Total histograms compared: 3569372
 - DQMHistoTests: [Total failures: 198](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_14_2_X_2024-10-30-1100+d54517/65378/validateJR.html)
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 3569154
 - DQMHistoTests: Total skipped: 20
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 46.016 KiB( 45 files compared)
 - DQMHistoSizes: changed ( 11634.0,... ): 2.876 KiB        Physics/NanoAODDQM
 - Checked 201 log files, 171 edm output root files, 46 DQM output files
 - [TriggerResults](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_14_2_X_2024-10-30-1100+d54517/65378/triggerResults): no differences found


## NANO Comparison Summary



**Summary**:
 - You potentially [removed 211 lines](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisonsNANO/CMSSW_14_2_X_2024-10-30-1100+d54517/65377/validateJR/logRootQA-events.log) from the logs
 - ROOTFileChecks: Some differences in event products or their sizes found
 - [Reco comparison results: 957 differences found in the comparisons](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisonsNANO/CMSSW_14_2_X_2024-10-30-1100+d54517/65377/validateJR.html)
 - DQMHistoTests: Total files compared: 21
 - DQMHistoTests: Total histograms compared: 55313
 - DQMHistoTests: [Total failures: 16](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisonsNANO/CMSSW_14_2_X_2024-10-30-1100+d54517/65377/validateJR.html)
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 55297
 - DQMHistoTests: Total skipped: 0
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 34.51200000000001 KiB( 20 files compared)
 - DQMHistoSizes: changed ( 2500.001,... ): 2.876 KiB        Physics/NanoAODDQM
 - Checked 102 log files, 58 edm output root files, 21 DQM output files
 - [TriggerResults](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisonsNANO/CMSSW_14_2_X_2024-10-30-1100+d54517/65377/triggerResults): no differences found

**Nano size comparison Summary**:

| Sample | kb/ev | ref kb/ev | diff kb/ev | ev/s/thd | ref ev/s/thd | diff rate | mem/thd | ref mem/thd |
| --- | --- | --- | --- | --- | --- | --- | --- | --- |
| 2500.001 | 3.039 | 3.012 | 0.027 ( +0.9% ) | 2.77 | 2.98 | -7.0% | 2.489 | 2.356 |
| 2500.002 | 3.153 | 3.124 | 0.028 ( +0.9% ) | 2.47 | 2.66 | -7.0% | 2.925 | 2.795 |
| 2500.003 | 3.093 | 3.065 | 0.027 ( +0.9% ) | 2.60 | 2.78 | -6.4% | 2.923 | 2.766 |
| 2500.011 | 1.603 | 1.591 | 0.012 ( +0.7% ) | 4.31 | 4.65 | -7.3% | 2.604 | 2.461 |
| 2500.012 | 2.134 | 2.113 | 0.022 ( +1.0% ) | 2.64 | 2.81 | -6.1% | 2.789 | 2.656 |
| 2500.013 | 1.956 | 1.936 | 0.019 ( +1.0% ) | 3.66 | 3.96 | -7.6% | 2.691 | 2.558 |
| 2500.021 | 0.022 | 0.022 | 0.000 ( +0.0% ) | 0.84 | 0.92 | -8.4% | 2.561 | 2.426 |
| 2500.022 | 0.022 | 0.022 | 0.000 ( +0.0% ) | 0.80 | 0.89 | -10.3% | 2.559 | 2.410 |
| 2500.023 | 0.022 | 0.022 | -0.000 ( -0.2% ) | 0.81 | 0.88 | -8.3% | 2.426 | 2.280 |
| 2500.024 | 0.022 | 0.022 | 0.000 ( +0.0% ) | 0.62 | 0.68 | -9.4% | 2.640 | 2.510 |
| 2500.031 | 0.035 | 0.035 | 0.000 ( +0.0% ) | 0.74 | 0.83 | -10.9% | 2.641 | 2.490 |
| 2500.032 | 0.036 | 0.036 | 0.000 ( +0.0% ) | 0.74 | 0.83 | -11.4% | 2.582 | 2.456 |
| 2500.033 | 0.037 | 0.037 | 0.000 ( +0.1% ) | 0.69 | 0.75 | -8.1% | 2.667 | 2.531 |
| 2500.034 | 0.036 | 0.036 | 0.000 ( +0.1% ) | 0.68 | 0.76 | -10.8% | 2.650 | 2.510 |
| 2500.101 | 2.776 | 2.766 | 0.010 ( +0.4% ) | 7.11 | 7.57 | -6.1% | 2.596 | 2.589 |
| 2500.111 | 1.426 | 1.419 | 0.008 ( +0.5% ) | 16.23 | 17.34 | -6.4% | 2.298 | 2.294 |
| 2500.112 | 1.848 | 1.838 | 0.010 ( +0.5% ) | 12.71 | 14.21 | -10.6% | 2.389 | 2.370 |
| 2500.131 | 0.747 | 0.747 | 0.000 ( +0.0% ) | 17.05 | 18.23 | -6.5% | 1.529 | 1.525 |
| 2500.201 | 2.605 | 2.595 | 0.010 ( +0.4% ) | 5.95 | 6.39 | -7.0% | 2.161 | 2.157 |
| 2500.211 | 1.766 | 1.757 | 0.008 ( +0.5% ) | 14.18 | 14.99 | -5.4% | 2.368 | 2.367 |
| 2500.212 | 2.162 | 2.152 | 0.009 ( +0.4% ) | 11.60 | 12.71 | -8.8% | 2.452 | 2.451 |
| 2500.221 | 2.016 | 2.016 | 0.000 ( +0.0% ) | 6.16 | 6.53 | -5.7% | 2.083 | 2.080 |
| 2500.222 | 3.403 | 3.392 | 0.011 ( +0.3% ) | 5.76 | 6.23 | -7.5% | 2.168 | 2.165 |
| 2500.223 | 9.368 | 9.357 | 0.011 ( +0.1% ) | 1.72 | 1.80 | -4.5% | 2.245 | 2.238 |
| 2500.224 | 6.051 | 6.040 | 0.011 ( +0.2% ) | 0.57 | 0.59 | -3.8% | 2.233 | 2.182 |
| 2500.225 | 6.098 | 6.072 | 0.026 ( +0.4% ) | 0.54 | 0.56 | -4.2% | 2.448 | 2.260 |
| 2500.226 | 3.097 | 3.086 | 0.011 ( +0.4% ) | 6.07 | 6.39 | -5.0% | 2.167 | 2.163 |
| 2500.227 | 1.437 | 1.437 | 0.000 ( +0.0% ) | 11.00 | 11.35 | -3.0% | 1.435 | 1.442 |
| 2500.231 | 1.429 | 1.429 | 0.000 ( +0.0% ) | 11.84 | 12.52 | -5.4% | 2.270 | 2.266 |
| 2500.232 | 2.421 | 2.412 | 0.009 ( +0.4% ) | 10.83 | 11.71 | -7.5% | 2.368 | 2.361 |
| 2500.233 | 4.913 | 4.904 | 0.009 ( +0.2% ) | 2.61 | 2.66 | -1.9% | 2.433 | 2.432 |
| 2500.234 | 3.679 | 3.670 | 0.009 ( +0.3% ) | 0.73 | 0.76 | -3.9% | 2.401 | 2.156 |
| 2500.235 | 3.709 | 3.689 | 0.021 ( +0.6% ) | 0.70 | 0.72 | -3.2% | 2.601 | 2.204 |
| 2500.236 | 2.211 | 2.202 | 0.009 ( +0.4% ) | 11.53 | 12.08 | -4.5% | 2.360 | 2.353 |
| 2500.237 | 1.016 | 1.016 | 0.000 ( +0.0% ) | 16.22 | 17.23 | -5.9% | 1.470 | 1.448 |
| 2500.241 | 9.404 | 9.404 | 0.000 ( +0.0% ) | 3.43 | 3.68 | -6.8% | 1.929 | 1.928 |
| 2500.242 | 10.331 | 10.331 | 0.000 ( +0.0% ) | 0.82 | 0.88 | -6.4% | 1.718 | 1.713 |
| 2500.243 | 2.712 | 2.712 | 0.000 ( +0.0% ) | 8.08 | 8.48 | -4.8% | 1.083 | 1.078 |
| 2500.244 | 485.976 | 485.976 | 0.000 ( +0.0% ) | 0.52 | 0.57 | -7.7% | 1.680 | 1.360 |
| 2500.245 | 823.202 | 823.202 | 0.000 ( +0.0% ) | 0.70 | 0.75 | -7.0% | 1.678 | 1.685 |
| 2500.901 | 1.777 | 1.777 | 0.000 ( +0.0% ) | 19.81 | 21.54 | -8.1% | 1.422 | 1.421 |
| 2500.902 | 1.626 | 1.626 | 0.000 ( +0.0% ) | 20.59 | 22.05 | -6.6% | 1.329 | 1.324 |
| 2500.911 | 13.995 | 13.995 | 0.000 ( +0.0% ) | 3.16 | 3.66 | -13.5% | 1.098 | 1.095 |
| 2500.912 | 0.199 | 0.310 | -0.111 ( -35.7% ) | 1.12 | 1.84 | -38.8% | 0.980 | 0.972 |
| 2500.913 | 0.110 | 0.110 | 0.000 ( +0.0% ) | 1.13 | 1.16 | -2.3% | 0.983 | 0.981 |

hqucms commented:
test parameters:
 - pull_request = https://github.com/cms-data/RecoBTag-Combined/pull/62
 - enable = nano

hqucms commented:
please test

cmsbuild commented:
+1

**Size**: This PR adds an extra 16KB to repository
**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-d54517/42533/summary.html
**COMMIT**: 243edc69092488ae4ed876479364b2cefa979658
**CMSSW**: CMSSW_14_2_X_2024-10-31-2300/el8_amd64_gcc12
**Additional Tests**: NANO
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week1/cms-sw/cmssw/46523/42533/install.sh` to create a dev area with all the needed externals and cmssw changes.



The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
 - @hjbossi cms-sw/cmssw#46537

You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-d54517/42533/git-recent-commits.json
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-d54517/42533/git-merge-result


## Comparison Summary



**Summary**:
 - You potentially [added 130 lines](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_14_2_X_2024-10-31-2300+d54517/65414/validateJR/logRootQA-events.log) to the logs
 - ROOTFileChecks: Some differences in event products or their sizes found
 - [Reco comparison results: 2786 differences found in the comparisons](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_14_2_X_2024-10-31-2300+d54517/65414/validateJR.html)
 - DQMHistoTests: Total files compared: 46
 - DQMHistoTests: Total histograms compared: 3569567
 - DQMHistoTests: [Total failures: 42](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_14_2_X_2024-10-31-2300+d54517/65414/validateJR.html)
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 3569505
 - DQMHistoTests: Total skipped: 20
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 46.016 KiB( 45 files compared)
 - DQMHistoSizes: changed ( 11634.0,... ): 2.876 KiB        Physics/NanoAODDQM
 - Checked 201 log files, 171 edm output root files, 46 DQM output files
 - [TriggerResults](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_14_2_X_2024-10-31-2300+d54517/65414/triggerResults): no differences found


## NANO Comparison Summary



**Summary**:
 - You potentially [removed 726 lines](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisonsNANO/CMSSW_14_2_X_2024-10-31-2300+d54517/65413/validateJR/logRootQA-events.log) from the logs
 - ROOTFileChecks: Some differences in event products or their sizes found
 - [Reco comparison results: 952 differences found in the comparisons](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisonsNANO/CMSSW_14_2_X_2024-10-31-2300+d54517/65413/validateJR.html)
 - DQMHistoTests: Total files compared: 21
 - DQMHistoTests: Total histograms compared: 55458
 - DQMHistoTests: [Total failures: 0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisonsNANO/CMSSW_14_2_X_2024-10-31-2300+d54517/65413/validateJR.html)
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 55458
 - DQMHistoTests: Total skipped: 0
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 34.51200000000001 KiB( 20 files compared)
 - DQMHistoSizes: changed ( 2500.001,... ): 2.876 KiB        Physics/NanoAODDQM
 - Checked 102 log files, 58 edm output root files, 21 DQM output files
 - [TriggerResults](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisonsNANO/CMSSW_14_2_X_2024-10-31-2300+d54517/65413/triggerResults): no differences found

**Nano size comparison Summary**:

| Sample | kb/ev | ref kb/ev | diff kb/ev | ev/s/thd | ref ev/s/thd | diff rate | mem/thd | ref mem/thd |
| --- | --- | --- | --- | --- | --- | --- | --- | --- |
| 2500.001 | 3.052 | 3.024 | 0.028 ( +0.9% ) | 2.74 | 2.95 | -7.2% | 2.500 | 2.365 |
| 2500.002 | 3.164 | 3.138 | 0.027 ( +0.9% ) | 2.46 | 2.61 | -5.8% | 2.521 | 2.802 |
| 2500.003 | 3.105 | 3.077 | 0.028 ( +0.9% ) | 2.58 | 2.72 | -5.2% | 2.537 | 2.771 |
| 2500.011 | 1.621 | 1.605 | 0.015 ( +1.0% ) | 4.23 | 4.52 | -6.4% | 2.597 | 2.465 |
| 2500.012 | 2.153 | 2.131 | 0.021 ( +1.0% ) | 2.56 | 2.69 | -4.8% | 2.411 | 2.658 |
| 2500.013 | 1.975 | 1.954 | 0.021 ( +1.1% ) | 3.55 | 3.83 | -7.4% | 2.672 | 2.559 |
| 2500.021 | 0.022 | 0.022 | 0.000 ( +0.0% ) | 0.89 | 0.94 | -5.7% | 2.564 | 2.419 |
| 2500.022 | 0.022 | 0.022 | 0.000 ( +0.0% ) | 0.84 | 0.90 | -6.2% | 2.561 | 2.419 |
| 2500.023 | 0.022 | 0.022 | 0.000 ( +0.0% ) | 0.85 | 0.89 | -5.1% | 2.410 | 2.274 |
| 2500.024 | 0.022 | 0.022 | 0.000 ( +0.0% ) | 0.64 | 0.68 | -6.9% | 2.645 | 2.503 |
| 2500.031 | 0.035 | 0.035 | 0.000 ( +0.0% ) | 0.75 | 0.81 | -7.3% | 2.617 | 2.475 |
| 2500.032 | 0.036 | 0.036 | 0.000 ( +0.0% ) | 0.77 | 0.84 | -8.9% | 2.585 | 2.441 |
| 2500.033 | 0.037 | 0.037 | 0.000 ( +0.0% ) | 0.70 | 0.77 | -8.8% | 2.669 | 2.526 |
| 2500.034 | 0.036 | 0.036 | 0.000 ( +0.0% ) | 0.71 | 0.77 | -8.0% | 2.648 | 2.504 |
| 2500.101 | 2.788 | 2.778 | 0.009 ( +0.3% ) | 6.77 | 7.15 | -5.4% | 2.618 | 2.609 |
| 2500.111 | 1.447 | 1.439 | 0.008 ( +0.6% ) | 13.27 | 13.85 | -4.2% | 2.321 | 2.312 |
| 2500.112 | 1.867 | 1.857 | 0.010 ( +0.5% ) | 11.37 | 11.98 | -5.0% | 2.385 | 2.391 |
| 2500.131 | 0.747 | 0.747 | 0.000 ( +0.0% ) | 17.41 | 18.33 | -5.0% | 1.466 | 1.501 |
| 2500.201 | 2.617 | 2.608 | 0.009 ( +0.3% ) | 5.67 | 6.06 | -6.4% | 2.168 | 2.176 |
| 2500.211 | 1.786 | 1.779 | 0.008 ( +0.4% ) | 11.63 | 12.15 | -4.3% | 2.379 | 2.374 |
| 2500.212 | 2.182 | 2.173 | 0.009 ( +0.4% ) | 10.13 | 10.34 | -2.0% | 2.466 | 2.465 |
| 2500.221 | 2.028 | 2.028 | 0.000 ( +0.0% ) | 5.98 | 6.23 | -3.9% | 2.084 | 2.094 |
| 2500.222 | 3.415 | 3.404 | 0.011 ( +0.3% ) | 5.65 | 5.86 | -3.7% | 2.186 | 2.184 |
| 2500.223 | 9.380 | 9.369 | 0.011 ( +0.1% ) | 1.72 | 1.79 | -4.2% | 2.205 | 2.255 |
| 2500.224 | 6.063 | 6.053 | 0.011 ( +0.2% ) | 0.57 | 0.59 | -3.2% | 2.096 | 2.244 |
| 2500.225 | 6.111 | 6.084 | 0.026 ( +0.4% ) | 0.54 | 0.56 | -3.8% | 2.008 | 2.319 |
| 2500.226 | 3.109 | 3.098 | 0.011 ( +0.4% ) | 5.77 | 6.06 | -4.7% | 2.178 | 2.181 |
| 2500.227 | 1.437 | 1.437 | 0.000 ( +0.0% ) | 11.22 | 11.92 | -5.9% | 1.431 | 1.434 |
| 2500.231 | 1.449 | 1.449 | 0.000 ( +0.0% ) | 9.59 | 10.28 | -6.7% | 1.725 | 2.278 |
| 2500.232 | 2.441 | 2.432 | 0.009 ( +0.4% ) | 9.25 | 9.71 | -4.7% | 2.372 | 2.376 |
| 2500.233 | 4.933 | 4.924 | 0.009 ( +0.2% ) | 2.56 | 2.66 | -3.8% | 2.437 | 2.444 |
| 2500.234 | 3.700 | 3.690 | 0.009 ( +0.3% ) | 0.73 | 0.76 | -3.7% | 2.168 | 2.404 |
| 2500.235 | 3.730 | 3.710 | 0.021 ( +0.6% ) | 0.70 | 0.71 | -1.5% | 2.353 | 2.468 |
| 2500.236 | 2.232 | 2.222 | 0.009 ( +0.4% ) | 9.92 | 10.31 | -3.8% | 2.378 | 2.371 |
| 2500.237 | 1.016 | 1.016 | 0.000 ( +0.0% ) | 16.57 | 17.42 | -4.9% | 1.447 | 1.456 |
| 2500.241 | 9.404 | 9.404 | 0.000 ( +0.0% ) | 3.61 | 3.84 | -6.1% | 1.914 | 1.923 |
| 2500.242 | 10.331 | 10.331 | 0.000 ( +0.0% ) | 0.84 | 0.91 | -7.1% | 1.708 | 1.700 |
| 2500.243 | 2.712 | 2.712 | 0.000 ( +0.0% ) | 7.97 | 8.41 | -5.2% | 1.077 | 1.076 |
| 2500.244 | 485.976 | 485.976 | 0.000 ( +0.0% ) | 0.54 | 0.57 | -6.6% | 1.684 | 1.680 |
| 2500.245 | 823.202 | 823.202 | 0.000 ( +0.0% ) | 0.70 | 0.76 | -7.3% | 1.661 | 1.663 |
| 2500.901 | 1.777 | 1.777 | 0.000 ( +0.0% ) | 20.10 | 21.57 | -6.8% | 1.145 | 1.416 |
| 2500.902 | 1.626 | 1.626 | 0.000 ( +0.0% ) | 20.72 | 22.24 | -6.8% | 1.086 | 1.323 |
| 2500.911 | 13.995 | 13.995 | 0.000 ( +0.0% ) | 3.81 | 4.00 | -4.8% | 1.090 | 1.088 |
| 2500.912 | 0.150 | 0.199 | -0.049 ( -24.7% ) | 1.22 | 1.51 | -19.2% | 0.968 | 0.976 |
| 2500.913 | 0.110 | 0.110 | 0.000 ( +0.0% ) | 1.04 | 1.18 | -11.9% | 0.976 | 0.982 |

hqucms commented:
+1

jfernan2 commented:
+1

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next master IBs (tests are also fine). This pull request will now be reviewed by the release team before it's merged. @sextonkennedy, @antoniovilela, @rappoccio, @mandrenguyen (and backports should be raised in the release meeting by the corresponding L2)
**Notice** This PR was tested with additional Pull Request(s), please also merge them if necessary: cms-data/RecoBTag-Combined#62

mandrenguyen commented:
+1

smuzaffar commented:
@colizz , there is a request to backport [RecoBTag-Combined tag V01-25-00](https://github.com/cms-sw/cmsdist/pull/9505) to 14.1.X. Do we need a backport of this PR to also go with https://github.com/cms-sw/cmsdist/pull/9505

hqucms commented:
@smuzaffar No, the changes introduced in V01-25-00 (i.e., https://github.com/cms-data/RecoBTag-Combined/pull/63) is unrelated to this PR.

smuzaffar commented:
what about changes in https://github.com/cms-data/RecoBTag-Combined/pull/62 ? https://github.com/cms-sw/cmsdist/pull/9505 include changes from both  https://github.com/cms-data/RecoBTag-Combined/pull/62 and  https://github.com/cms-data/RecoBTag-Combined/pull/63 . So in order to integrate https://github.com/cms-sw/cmsdist/pull/9505  in to 14.1.X, do we need cmssw update too?

hqucms commented:
No -- https://github.com/cms-data/RecoBTag-Combined/pull/62 only adds new models.
